impl/src/main/java/com/okta/sdk/impl/resource/DefaultUserBuilder.java

```diff
@@ -168,28 +170,66 @@ private User build(Client client) {
 
         userProfile.putAll(customProfileAttributes);
 
-        if (password != null && password.length > 0 || Strings.hasText(securityQuestion)) {
-            UserCredentials credentials = client.instantiate(UserCredentials.class);
-            user.setCredentials(credentials);
+        // security question
+        if (Strings.hasText(securityQuestion)) {
+            RecoveryQuestionCredential question = client.instantiate(RecoveryQuestionCredential.class);
+            question.setQuestion(securityQuestion);
+            question.setAnswer(securityQuestionAnswer);
+            createCredentialsIfNeeded(user, client).setRecoveryQuestion(question);
+        }
 
-            if (Strings.hasText(securityQuestion)) {
-                RecoveryQuestionCredential question = client.instantiate(RecoveryQuestionCredential.class);
-                question.setQuestion(securityQuestion);
-                question.setAnswer(securityQuestionAnswer);
-                credentials.setRecoveryQuestion(question);
-            }
+        // user password
+        if (password != null && password.length > 0) {
+            PasswordCredential passwordCredential = client.instantiate(PasswordCredential.class);
+            createCredentialsIfNeeded(user, client).setPassword(passwordCredential.setValue(password));
+        }
 
-            if (password != null && password.length > 0) {
-                PasswordCredential passwordCredential = client.instantiate(PasswordCredential.class);
-                credentials.setPassword(passwordCredential.setValue(password));
-            }
+        // password import
+        if (passwordHashProperties != null) {
```

Comment:
Thoughts on erroring if both a password and a hash are specified? Unlikely to happen, but looks like it would silently overwrite the PasswordCredential created above.